tools/wasm-sourcemap.py

```diff
@@ -231,15 +233,183 @@ def extract_comp_dir_map(text):
+  #    DW_AT_high_pc. This represents an inlined function. We parse
+  #    DW_AT_abstract_origin for the original function name.
+  # 0x0000011a:   DW_TAG_inlined_subroutine
+  #                 DW_AT_abstract_origin  (0x000000da "_ZN7MyClass3barEv")
+  #                 DW_AT_low_pc  (0x00000078)
+  #                 DW_AT_high_pc  (0x00000083)
+  #                 ...
+
+  func_ranges = []
+  dw_tags = re.split(r'\r?\n(?=0x[0-9a-f]+:)', text)
+
+  def get_name_from_tag(tag):
+    m = re.search(r'DW_AT_linkage_name\s+\("([^"]+)"\)', tag)
+    if m:
+      return m.group(1)
+    m = re.search(r'DW_AT_name\s+\("([^"]+)"\)', tag)
+    if m:
+      return m.group(1)
+    # If name is missing, check for DW_AT_specification annotation
+    m = re.search(r'DW_AT_specification\s+\(0x[0-9a-f]+\s+"([^"]+)"\)', tag)
+    if m:
+      return m.group(1)
+    return None
+
+  for tag in dw_tags:
+    is_subprogram = re.search(r"0x[0-9a-f]+:\s+DW_TAG_subprogram", tag)
+    is_inlined = re.search(r"0x[0-9a-f]+:\s+DW_TAG_inlined_subroutine", tag)
+    if is_subprogram or is_inlined:
+      name = None
+      low_pc = None
+      high_pc = None
+      m = re.search(r'DW_AT_low_pc\s+\(0x([0-9a-f]+)\)', tag)
+      if m:
+        low_pc = int(m.group(1), 16)
+      m = re.search(r'DW_AT_high_pc\s+\(0x([0-9a-f]+)\)', tag)
+      if m:
+        high_pc = int(m.group(1), 16)
+      if is_subprogram:
+        name = get_name_from_tag(tag)
+      else: # is_inlined
+        m = re.search(r'DW_AT_abstract_origin\s+\(0x[0-9a-f]+\s+"([^"]+)"\)', tag)
+        if m:
+          name = m.group(1)
+      if name and low_pc is not None and high_pc is not None:
+        func_ranges.append(FuncRange(name, low_pc, high_pc))
+
+  # Demangle names
+  all_names = [item.name for item in func_ranges]
+  demangled_map = demangle_names(all_names)
+  for func_range in func_ranges:
+    if func_range.name in demangled_map:
+      func_range.name = demangled_map[func_range.name]
+
+  # To correctly identify the innermost function for a given address,
+  # func_ranges is sorted primarily by low_pc in ascending order and secondarily
+  # by high_pc in descending order. This ensures that for overlapping ranges,
+  # the more specific (inner) range appears later in the list.
+  func_ranges.sort(key=lambda item: (item.low_pc, -item.high_pc))
+  return func_ranges
+
+
+# Returns true if the given llvm-dwarfdump has --filter-child-tags / -t option
+def has_filter_child_tag_option(dwarfdump):
+  # To check if --filter-child-tags / -t option is available, run
+  # `llvm-dwarfdump -t`. If it is available, it will print to stderr:
+  #   ... for the -t option: requires a value!
+  # If not, it will print:
+  #   ... Unknown command line argument '-t'.
+  try:
+    process = Popen([dwarfdump, '-t'], stdout=PIPE, stderr=PIPE, text=True)
+    _, err = process.communicate()
+    return 'requires a value' in err
+  except OSError:
+    return False
```

Comment:
Again, use `check_call` to avoid direct usage of subprocess.

Reply:
This part was removed in https://github.com/emscripten-core/emscripten/pull/25870/commits/d36dad5f1291eb38b8637133683b883184e11fed